### PR TITLE
fix(DB/creature) Slain Auchenai Warrior & Slain Sha'tar Vindicator

### PR DESCRIPTION
**Pull request by Pyromanihack**
## Changes Proposed:
-  Pin the "1" bit on the flag of the "CALL_KILLEDMONSTER" action
-  Add more spawn of Slain Auchenai Warrior and Slain Sha'tar Vindicator (in total we have 15 Warrior and 13 Vindicator)

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/13670
- Closes https://github.com/chromiecraft/chromiecraft/issues/4385

## Tests Performed:

- 
- 

## How to Test the Changes:

1. .quest add 10913
2. .go cr 1977353
3. Check if the burned corpse gives you only one count for the quest
4. Look around and see in the corpse are spawned in addictional points

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Gultask** [comment]
Hey, glad to see another PR! :)
What's the source for the new spawns?

**Comment by Pyromanihack** [comment]
> Hey, glad to see another PR! :) What's the source for the new spawns?

I took them from wowhead
https://www.wowhead.com/tbc/npc=21859/slain-shatar-vindicator
https://www.wowhead.com/tbc/npc=21846/slain-auchenai-warrior

**Comment by Gultask** [comment]
Ah, there's a problem there. We don't take values from Wowhead. That's because most values we take from either other DBs or, preferrably, from sniffs. Wowhead values are very prone to errors while sniff values can be close to perfect.

So I went to take a look and it seems like TrinityCore does have 36 spawns of these 2 NPCs, so I exported it. I haven't tested yet other than running it 3 times in a SQL Client.

There are 3 things here:
1. I'm using id2 alongside id1, what this means is that there will be a random chance of either IDs spawning. TC uses pooling for these, and I believe this will achieve the same result.
2. I like to use ZoneId for organization :)
3. I'm using SET @GUID to make a constant for easier readability. It also makes it easier to change GUIDs if something unexpected occurs. You can use https://github.com/azerothcore/Unused-Guid-Search this to search for unused consecutive guids.

Could you test it out to see if it works with these new spawns?

```
SET @GUID = 1977353;

DELETE FROM `creature` WHERE `id1` IN (21846, 21859) AND `id2` IN (21846, 21859) AND `guid` BETWEEN @GUID+0 AND @GUID+35;
INSERT INTO `creature` (`guid`, `id1`, `id2`, `map`, `zoneId`, `areaId`, `spawnMask`, `phaseMask`, `position_x`, `position_y`, `position_z`, `orientation`, `spawntimesecs`, `MovementType`) VALUES
(@GUID+0 , 21846, 21859, 530, 3519, 0, 1, 1, -3780.56, 5215.07, -23.1934, 0.0349066, 7, 0),
(@GUID+1 , 21846, 21859, 530, 3519, 0, 1, 1, -3777.38, 5184.86, -24.4552, 3.9968, 7, 0),
(@GUID+2 , 21846, 21859, 530, 3519, 0, 1, 1, -3772.17, 5211.63, -22.8059, 5.49779, 7, 0),
(@GUID+3 , 21846, 21859, 530, 3519, 0, 1, 1, -3770.83, 5121.01, -21.4213, 0.959931, 7, 0),
(@GUID+4 , 21846, 21859, 530, 3519, 0, 1, 1, -3766.29, 5062.47, -18.9817, 3.57792, 7, 0),
(@GUID+5 , 21846, 21859, 530, 3519, 0, 1, 1, -3764.99, 5095.88, -19.0595, 0.122173, 7, 0),
(@GUID+6 , 21846, 21859, 530, 3519, 0, 1, 1, -3762.46, 5171.85, -22.9264, 1.0821, 7, 0),
(@GUID+7 , 21846, 21859, 530, 3519, 0, 1, 1, -3759.02, 5140.41, -22.8351, 4.81711, 7, 0),
(@GUID+8 , 21846, 21859, 530, 3519, 0, 1, 1, -3758.07, 5261.12, -15.3995, 2.77507, 7, 0),
(@GUID+9 , 21846, 21859, 530, 3519, 0, 1, 1, -3753.9, 5294.92, -16.9184, 1.43117, 7, 0),
(@GUID+10, 21846, 21859, 530, 3519, 0, 1, 1, -3753.17, 5078.65, -18.2274, 0.872665, 7, 0),
(@GUID+11, 21846, 21859, 530, 3519, 0, 1, 1, -3750.03, 5125.17, -21.9827, 5.35816, 7, 0),
(@GUID+12, 21846, 21859, 530, 3519, 0, 1, 1, -3749.58, 5195.27, -22.6601, 3.68264, 7, 0),
(@GUID+13, 21846, 21859, 530, 3519, 0, 1, 1, -3741.7, 5160.73, -22.2081, 4.50295, 7, 0),
(@GUID+14, 21846, 21859, 530, 3519, 0, 1, 1, -3741.05, 5171.27, -21.8304, 2.14675, 7, 0),
(@GUID+15, 21846, 21859, 530, 3519, 0, 1, 1, -3739.48, 5271.71, -16.0977, 0.523599, 7, 0),
(@GUID+16, 21846, 21859, 530, 3519, 0, 1, 1, -3734.78, 5327.33, -13.0874, 3.9619, 7, 0),
(@GUID+17, 21846, 21859, 530, 3519, 0, 1, 1, -3733.21, 5061.52, -17.6918, 0.628319, 7, 0),
(@GUID+18, 21846, 21859, 530, 3519, 0, 1, 1, -3731.54, 5160.66, -22.2169, 3.4383, 7, 0),
(@GUID+19, 21846, 21859, 530, 3519, 0, 1, 1, -3726.95, 5263.67, -17.9404, 3.66519, 7, 0),
(@GUID+20, 21846, 21859, 530, 3519, 0, 1, 1, -3721.04, 5275.86, -18.5205, 1.25664, 7, 0),
(@GUID+21, 21846, 21859, 530, 3519, 0, 1, 1, -3720.57, 5205.87, -20.2053, 2.53073, 7, 0),
(@GUID+22, 21846, 21859, 530, 3519, 0, 1, 1, -3718.09, 5217.56, -21.3539, 2.75762, 7, 0),
(@GUID+23, 21846, 21859, 530, 3519, 0, 1, 1, -3713.32, 5257.14, -20.3728, 3.82227, 7, 0),
(@GUID+24, 21846, 21859, 530, 3519, 0, 1, 1, -3711.44, 5290.39, -18.9303, 0.0698132, 7, 0),
(@GUID+25, 21846, 21859, 530, 3519, 0, 1, 1, -3691.57, 5220.81, -21.1431, 2.54818, 7, 0),
(@GUID+26, 21846, 21859, 530, 3519, 0, 1, 1, -3689.51, 5295.31, -20.0296, 3.33358, 7, 0),
(@GUID+27, 21846, 21859, 530, 3519, 0, 1, 1, -3688.32, 5336.11, -15.1771, 3.735, 7, 0),
(@GUID+28, 21846, 21859, 530, 3519, 0, 1, 1, -3687.49, 5351.75, -13.172, 3.1765, 7, 0),
(@GUID+29, 21846, 21859, 530, 3519, 0, 1, 1, -3679.08, 5287.66, -20.6681, 0.0698132, 7, 0),
(@GUID+30, 21846, 21859, 530, 3519, 0, 1, 1, -3678.04, 5293.5, -20.3924, 1.78024, 7, 0),
(@GUID+31, 21846, 21859, 530, 3519, 0, 1, 1, -3676.77, 5334.48, -16.1081, 0.575959, 7, 0),
(@GUID+32, 21846, 21859, 530, 3519, 0, 1, 1, -3643.76, 5323.48, -18.2017, 4.08407, 7, 0),
(@GUID+33, 21846, 21859, 530, 3519, 0, 1, 1, -3640.03, 5318.27, -19.0845, 3.38594, 7, 0),
(@GUID+34, 21846, 21859, 530, 3519, 0, 1, 1, -3634.82, 5283.14, -21.9279, 4.5204, 7, 0),
(@GUID+35, 21846, 21859, 530, 3519, 0, 1, 1, -3783.05, 5215.14, -23.515, 1.76278, 7, 0);
```

**Comment by Pyromanihack** [comment]
It's a great explanation, and a very useful tool (https://github.com/azerothcore/Unused-Guid-Search).
What do you mean with "i pinged someone"?

**Comment by Gultask** [comment]
I pinged someone named GUID when I typed @ GUID without the spaces 😛 and probably gave them a notification.

File: data/sql/updates/pending_db_world/AnImproperBurial.sql [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+(1977371, 21846, 0, 0, 530, 0, 0, 1, 1, 1, -3642.40, 5322.93, -18.42, 0, 120, 0, 0, 1, 0, 0, 0, 0, 0, '', 0),
+(1977372, 21846, 0, 0, 530, 0, 0, 1, 1, 1, -3678.40, 5301.33, -19.96, 0, 120, 0, 0, 1, 0, 0, 0, 0, 0, '', 0),
+(1977373, 21846, 0, 0, 530, 0, 0, 1, 1, 1, -3692.8, 5225.73, -21.89, 0, 120, 0, 0, 1, 0, 0, 0, 0, 0, '', 0),
+(1977374, 21846, 0, 0, 530, 0, 0, 1, 1, 1, -3779.2, 5214.93, -23.13, 0, 120, 0, 0, 1, 0, 0, 0, 0, 0, '', 0),
+(1977375, 21846, 0, 0, 530, 0, 0, 1, 1, 1, -3757.59, 5074.53, -18.53, 0, 120, 0, 0, 1, 0, 0, 0, 0, 0, '', 0),
+(1977376, 21859, 0, 0, 530, 0, 0, 1, 1, 1, -3685.59, 5344.53, -14.36, 0, 120, 0, 0, 1, 0, 0, 0, 0, 0, '', 0),
+(1977377, 21859, 0, 0, 530, 0, 0, 1, 1, 1, -3678.40, 5333.73, -16.21, 0, 120, 0, 0, 1, 0, 0, 0, 0, 0, '', 0),
+(1977378, 21859, 0, 0, 530, 0, 0, 1, 1, 1, -3642.40, 5322.93, -18.42, 0, 120, 0, 0, 1, 0, 0, 0, 0, 0, '', 0),
+(1977379, 21859, 0, 0, 530, 0, 0, 1, 1, 1, -3678.40, 5301.33, -19.96, 0, 120, 0, 0, 1, 0, 0, 0, 0, 0, '', 0),
+(1977380, 21859, 0, 0, 530, 0, 0, 1, 1, 1, -3743.2, 5171.73, -21.93, 0, 120, 0, 0, 1, 0, 0, 0, 0, 0, '', 0);
```
Comment (Gultask) [inline data/sql/updates/pending_db_world/AnImproperBurial.sql:19]: We can't take values from Wowhead unless there's absolutely no other way to fix it.

```suggestion
-- Add spawns imported from TC
DELETE FROM `creature` WHERE `id1` IN (21846, 21859) AND `id2` IN (21846, 21859) AND `guid` BETWEEN @GUID+0 AND @GUID+35;
INSERT INTO `creature` (`guid`, `id1`, `id2`, `map`, `zoneId`, `areaId`, `spawnMask`, `phaseMask`, `position_x`, `position_y`, `position_z`, `orientation`, `spawntimesecs`, `MovementType`) VALUES
(@GUID+0 , 21846, 21859, 530, 3519, 0, 1, 1, -3780.56, 5215.07, -23.1934, 0.0349066, 7, 0),
(@GUID+1 , 21846, 21859, 530, 3519, 0, 1, 1, -3777.38, 5184.86, -24.4552, 3.9968, 7, 0),
(@GUID+2 , 21846, 21859, 530, 3519, 0, 1, 1, -3772.17, 5211.63, -22.8059, 5.49779, 7, 0),
(@GUID+3 , 21846, 21859, 530, 3519, 0, 1, 1, -3770.83, 5121.01, -21.4213, 0.959931, 7, 0),
(@GUID+4 , 21846, 21859, 530, 3519, 0, 1, 1, -3766.29, 5062.47, -18.9817, 3.57792, 7, 0),
(@GUID+5 , 21846, 21859, 530, 3519, 0, 1, 1, -3764.99, 5095.88, -19.0595, 0.122173, 7, 0),
(@GUID+6 , 21846, 21859, 530, 3519, 0, 1, 1, -3762.46, 5171.85, -22.9264, 1.0821, 7, 0),
(@GUID+7 , 21846, 21859, 530, 3519, 0, 1, 1, -3759.02, 5140.41, -22.8351, 4.81711, 7, 0),
(@GUID+8 , 21846, 21859, 530, 3519, 0, 1, 1, -3758.07, 5261.12, -15.3995, 2.77507, 7, 0),
(@GUID+9 , 21846, 21859, 530, 3519, 0, 1, 1, -3753.9, 5294.92, -16.9184, 1.43117, 7, 0),
(@GUID+10, 21846, 21859, 530, 3519, 0, 1, 1, -3753.17, 5078.65, -18.2274, 0.872665, 7, 0),
(@GUID+11, 21846, 21859, 530, 3519, 0, 1, 1, -3750.03, 5125.17, -21.9827, 5.35816, 7, 0),
(@GUID+12, 21846, 21859, 530, 3519, 0, 1, 1, -3749.58, 5195.27, -22.6601, 3.68264, 7, 0),
(@GUID+13, 21846, 21859, 530, 3519, 0, 1, 1, -3741.7, 5160.73, -22.2081, 4.50295, 7, 0),
(@GUID+14, 21846, 21859, 530, 3519, 0, 1, 1, -3741.05, 5171.27, -21.8304, 2.14675, 7, 0),
(@GUID+15, 21846, 21859, 530, 3519, 0, 1, 1, -3739.48, 5271.71, -16.0977, 0.523599, 7, 0),
(@GUID+16, 21846, 21859, 530, 3519, 0, 1, 1, -3734.78, 5327.33, -13.0874, 3.9619, 7, 0),
(@GUID+17, 21846, 21859, 530, 3519, 0, 1, 1, -3733.21, 5061.52, -17.6918, 0.628319, 7, 0),
(@GUID+18, 21846, 21859, 530, 3519, 0, 1, 1, -3731.54, 5160.66, -22.2169, 3.4383, 7, 0),
(@GUID+19, 21846, 21859, 530, 3519, 0, 1, 1, -3726.95, 5263.67, -17.9404, 3.66519, 7, 0),
(@GUID+20, 21846, 21859, 530, 3519, 0, 1, 1, -3721.04, 5275.86, -18.5205, 1.25664, 7, 0),
(@GUID+21, 21846, 21859, 530, 3519, 0, 1, 1, -3720.57, 5205.87, -20.2053, 2.53073, 7, 0),
(@GUID+22, 21846, 21859, 530, 3519, 0, 1, 1, -3718.09, 5217.56, -21.3539, 2.75762, 7, 0),
(@GUID+23, 21846, 21859, 530, 3519, 0, 1, 1, -3713.32, 5257.14, -20.3728, 3.82227, 7, 0),
(@GUID+24, 21846, 21859, 530, 3519, 0, 1, 1, -3711.44, 5290.39, -18.9303, 0.0698132, 7, 0),
(@GUID+25, 21846, 21859, 530, 3519, 0, 1, 1, -3691.57, 5220.81, -21.1431, 2.54818, 7, 0),
(@GUID+26, 21846, 21859, 530, 3519, 0, 1, 1, -3689.51, 5295.31, -20.0296, 3.33358, 7, 0),
(@GUID+27, 21846, 21859, 530, 3519, 0, 1, 1, -3688.32, 5336.11, -15.1771, 3.735, 7, 0),
(@GUID+28, 21846, 21859, 530, 3519, 0, 1, 1, -3687.49, 5351.75, -13.172, 3.1765, 7, 0),
(@GUID+29, 21846, 21859, 530, 3519, 0, 1, 1, -3679.08, 5287.66, -20.6681, 0.0698132, 7, 0),
(@GUID+30, 21846, 21859, 530, 3519, 0, 1, 1, -3678.04, 5293.5, -20.3924, 1.78024, 7, 0),
(@GUID+31, 21846, 21859, 530, 3519, 0, 1, 1, -3676.77, 5334.48, -16.1081, 0.575959, 7, 0),
(@GUID+32, 21846, 21859, 530, 3519, 0, 1, 1, -3643.76, 5323.48, -18.2017, 4.08407, 7, 0),
(@GUID+33, 21846, 21859, 530, 3519, 0, 1, 1, -3640.03, 5318.27, -19.0845, 3.38594, 7, 0),
(@GUID+34, 21846, 21859, 530, 3519, 0, 1, 1, -3634.82, 5283.14, -21.9279, 4.5204, 7, 0),
(@GUID+35, 21846, 21859, 530, 3519, 0, 1, 1, -3783.05, 5215.14, -23.515, 1.76278, 7, 0);
```

**Review by Gultask — CHANGES_REQUESTED** [review]

File: data/sql/updates/pending_db_world/AnImproperBurial.sql (hunk 1 above)
Comment (Pyromanihack) [reply data/sql/updates/pending_db_world/AnImproperBurial.sql:19]: SET @GUID = 1977353;

Should i put this line like you suggest earlier?

**Comment by Gultask** [comment]
Tested. The quest works and one spawn cannot be reused over and over. The Slain Sha'tar Vindicator has gossip but that doesn't need to be addressed in this PR, I think.

**Review by Gultask — APPROVED** [review]

**Pull request merged**